src/cdk/portal/portal.ts

```diff
@@ -227,6 +246,9 @@ export abstract class BasePortalOutlet implements PortalOutlet {
 
   abstract attachTemplatePortal<C>(portal: TemplatePortal<C>): EmbeddedViewRef<C>;
 
+  // @breaking-change 9.0.0 `attachDomPortal` to become a required method.
+  abstract attachDomPortal?(portal: DomPortal): any;
```

Comment:
This is not good enough for backwards compatibility, because TS still complains about the unimplemented abstract method. Instead maybe: `readonly attachDomPortal: null | (portal: DomPortal) => any = null;` (with a comment to make it a normal method in a future release).